### DISCUSSION TITLE
[5.3.0] How to get OpenStreetMap to work?

### DISCUSSION THREAD

**Discussion by vuongquoc** [LycheeOrg/Lychee#2344] [Q&A] [answered]
It was working just fine in 4.x.x but since i update to 5.x.x,
i can't get it working what/which-ever options i choose in backend...

It just shows the map without any images on locations...

How can i get it to work?

Thanks...

VuongQuoc

**Comment by ildyria** [comment]
Which version of 5.x.x are you using ? 
I did solve a few bugs on master, but we haven't had time to get them released (as there are pretty much nobody reviewing the PR...).
As far as I know, it is working fine for me locally.
Do know that the layout is broken for one of the map (I don't remember which one).

**Reply by vuongquoc** [reply]
Hi Sir,
I'm using 5.1.2

VuongQuoc

**Reply by vuongquoc** [reply]
updated to 5.2.2 but still not working though...
I've tried all the map options but not working...

**Reply by ildyria** [reply]
Sorry miss read.

**Reply by ildyria** [reply]
> It just shows the map without any images on locations...

Would it be possible for you to send me (benoit@viguier.nl) a DB dump so I can investigate this in more details?
That is really weird because it is working on my setup.

**Reply by vuongquoc** [reply]
![image](https://github.com/LycheeOrg/Lychee/assets/84169528/ec9f8cf8-b8ec-4be6-85f7-6fa5f9d121b6)
for example,
i uploaded an image from my pixel 8 pro which already turned on location.
But after upload, it do not read the location while it showed on the phone.

may be it's a problem on server side...
I'm trying to check in my cloud server and my php 8.3 options but the server is upgrading so can not check at the moment.

I'll update after checked on the server...

**Reply by d7415** [reply]
If uploading from the phone, the phone will often strip location data. Maybe try downloading it from Lychee and checking what's there, or uploading from a computer.

**Reply by vuongquoc** [reply]
I'm uploading from pc...

**Reply by ildyria** [reply]
Could you send me that picture by email (benoit@viguier.nl) with the altitude but no longitude & latitude ? 
I would like to investigate more on that.

**Comment by vuongquoc** [comment]
> Could you send me that picture by email ([benoit@viguier.nl](mailto:benoit@viguier.nl)) with the altitude but no longitude & latitude ? I would like to investigate more on that.

Hi sir,
i've just emailed you one of my pictures, please have a look when you have time!
no haste plz!

Thank you!

**Reply by ildyria** [reply, marked as answer]
Hi @vuongquoc so I got some good news, and some bad news.
The good news: uploading your image locally on my lychee.test does work.
![image](https://github.com/LycheeOrg/Lychee/assets/627094/a07ac587-0c27-4dcc-a1cb-7052a81f161f)
(and yes it displays on map, but I don't want to doxx you).

The bad news: there is something going wrong in the reading of the GPS data on your server side. >_<

**Comment by vuongquoc** [comment]
> Hi @vuongquoc so I got some good news, and some bad news. The good news: uploading your image locally on my lychee.test does work. ![image](https://private-user-images.githubusercontent.com/627094/327515934-a07ac587-0c27-4dcc-a1cb-7052a81f161f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3MTExNTMsIm5iZiI6MTcxNDcxMDg1MywicGF0aCI6Ii82MjcwOTQvMzI3NTE1OTM0LWEwN2FjNTg3LTBjMjctNGRjYy1hMWNiLTcwNTJhODFmMTYxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwM1QwNDM0MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMmE2ZTA2NGI2ZmJiZTQyYWM3ZWM0NjkzMGZhNGY1ODQ0MzYyNjdiNzQwMzY3MGQwYzRhZDA1MDQzNjU2YzFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Pe2-QCkVx7SZkta2uVs5EULgvPc9uXzwuo1t-sEiRLA) (and yes it displays on map, but I don't want to doxx you).
> 
> The bad news: there is something going wrong in the reading of the GPS data on your server side. >_<

Thank you sir, for your time.
I also think it's my server problem and it's upgrading so maybe i should wait until it fully updated.

VuongQuoc

**Reply by ildyria** [reply]
I would check if the php exif extension is installed and also the php imagik extension.

**Reply by vuongquoc** [reply]
Godaddy Cpanel Cloud Server are upgrading and i can not check in PHP options right now so have to wait weeks after they upgrading server... :)

Thanks for your advises!!!

VuongQuoc
